crypto/hmac_extra/hmac_test.cc

```diff
@@ -89,6 +90,69 @@ static const EVP_MD *GetDigest(const std::string &name) {
   return nullptr;
 }
 
+static bool RunHMACTestEVP(const std::vector<uint8_t> &key,
+                           const std::vector<uint8_t> &msg,
+                           const std::vector<uint8_t> &tag, const EVP_MD *md) {
+  bssl::UniquePtr<EVP_PKEY> pkey(
+      EVP_PKEY_new_mac_key(EVP_PKEY_HMAC, nullptr, key.data(), key.size()));
+  EXPECT_TRUE(pkey);
+
+  bssl::ScopedEVP_MD_CTX copy, mctx;
+  size_t len;
+  std::vector<uint8_t> actual;
+  EXPECT_TRUE(EVP_DigestSignInit(mctx.get(), nullptr, md, nullptr, pkey.get()));
+  // Make a copy we can test against later.
+  EXPECT_TRUE(EVP_MD_CTX_copy_ex(copy.get(), mctx.get()));
+  EXPECT_TRUE(EVP_DigestSignUpdate(mctx.get(), msg.data(), msg.size()));
+  EXPECT_TRUE(EVP_DigestSignFinal(mctx.get(), nullptr, &len));
+  actual.resize(len);
+  EXPECT_TRUE(EVP_DigestSignFinal(mctx.get(), actual.data(), &len));
+  actual.resize(len);
```

Comment:
In theory the size shouldn't change right?
```suggestion
  EXPECT_EQ(len.size(), len);
```